### DISCUSSION TITLE
nextjs.org/learn/dashboard-app/ might have problem and I'm not quite sure which issue category fits the problem to submit

### DISCUSSION THREAD

**Discussion by Ali-Milani-13** [vercel/next.js#68877] [Help]
### Summary

I want to inform Vercel about this part of the learning document: 

[navigating-between-pages](https://nextjs.org/learn/dashboard-app/navigating-between-pages)

if you use the code from last part, it will give an error message with this message: 

> Error: Unsupported Server Component type: undefined

it also gives me a message to update the nextjs Canary with this message:
 
`Next.js (15.0.0-canary.56) is outdated
`

and when I try to update my Nextjs for this tutorial, I get this message from pnpm:

![Screenshot 2024-08-14 124534](https://github.com/user-attachments/assets/db15a0a7-46c3-43a3-8c73-e90337d951a1)

```

'use client';
 
import {
  UserGroupIcon,
  HomeIcon,
  DocumentDuplicateIcon,
} from '@heroicons/react/24/outline';
import Link from 'next/link';
import { usePathname } from 'next/navigation';
import clsx from 'clsx';
 
const links = [
  { name: 'Home', href: '/dashboard', icon: HomeIcon },
  {
    name: 'Invoices',
    href: '/dashboard/invoices',
    icon: DocumentDuplicateIcon,
  },
  { name: 'Customers', href: '/dashboard/customers', icon: UserGroupIcon },
];
 
export default function NavLinks() {
  const pathname = usePathname();
 
  return (
    <>
      {links.map((link) => {
        const LinkIcon = link.icon;
        return (
          <Link
            key={link.name}
            href={link.href}
            className={clsx(
              'flex h-[48px] grow items-center justify-center gap-2 rounded-md bg-gray-50 p-3 text-sm font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3',
              {
                'bg-sky-100 text-blue-600': pathname === link.href,
              },
            )}
          >
            <LinkIcon className="w-6" />
            <p className="hidden md:block">{link.name}</p>
          </Link>
        );
      })}
    </>
  );
}
```

### Additional information

```js
please consider tell me where I should submit this issue because I'm not sure about which category it fits
```

**Comment by bartbuba** [comment]
bro I have the same error. Did you solve it?

**Reply by Ali-Milani-13** [reply]
I think there's a problem with pnmp package manager. 

I deleted the whole project and used npm instead, to set it up. it somehow worked but since I'm not sure if it is 100% for pnpm or some sort of bug with Nextjs, I'm not going to close this issue for now. 

try doing the same, it might work for you as well.
